static/modes/cppcircle-mode.js

```diff
@@ -0,0 +1,85 @@
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+'use strict';
+var $ = require('jquery');
+var monaco = require('monaco-editor');
+var cpp = require('monaco-editor/esm/vs/basic-languages/cpp/cpp');
+
+// We need to create a new definition for cpp so we can remove invalid keywords
+
+function definition() {
+    var cppcircle = $.extend(true, {}, cpp.language); // deep copy
+
+    function removeKeyword(keyword) {
+        var index = cppcircle.keywords.indexOf(keyword);
+        if (index > -1) {
+            cppcircle.keywords.splice(index, 1);
+        }
+    }
+
+    function removeKeywords(keywords) {
+        for (var i = 0; i < keywords.length; ++i) {
+            removeKeyword(keywords[i]);
+        }
+    }
+
+    function addKeywords(keywords) {
+        // (Ruben) Done one by one as if you just push them all, Monaco complains that they're not strings, but as
+        // far as I can tell, they indeed are all strings. This somehow fixes it. If you know how to fix it, plz go
+        for (var i = 0; i < keywords.length; ++i) {
+            cppcircle.keywords.push(keywords[i]);
+        }
+    }
+
+    // We remove everything that's not an identifier, underscore reserved name and not an official C++ keyword...
+    // Regarding #617, final is a identifier with special meaning, not a fully qualified keyword
+    removeKeywords(['abstract', 'amp', 'array', 'cpu', 'delegate', 'each', 'event', 'finally', 'gcnew',
+        'generic', 'in', 'initonly', 'interface', 'interior_ptr', 'internal', 'literal', 'partial', 'pascal',
+        'pin_ptr', 'property', 'ref', 'restrict', 'safe_cast', 'sealed', 'title_static', 'where']);
+
+    addKeywords(['@array', '@attribute', '@base_count', '@base_offset', '@base_offsets', '@base_type',
```

Comment:
I'll go play with the editor to see why the `@` keywords aren't working. ( https://microsoft.github.io/monaco-editor/monarch.html if you want to try too! )

Reply:
Hack does work. Thanks!
```js
    // Hack to put `@...` keywords in place
    cppcircle.tokenizer.root.unshift(
        [/@\w+/, {cases: {'@keywords': {token: 'keyword.$0'}}}]
    );
```